avm/res/maintenance/configuration-assignment/main.bicep

```diff
@@ -0,0 +1,109 @@
+metadata name = 'Maintenance Configuration Assignments'
+metadata description = 'This module deploys a Maintenance Configuration Assignment.'
+
+// ============== //
+//   Parameters   //
+// ============== //
+
+@description('Required. Maintenance configuration assignment Name.')
+param name string
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Required. The maintenance configuration resource ID.')
+param maintenanceConfigurationResourceId string
+
+@description('Conditional. The unique virtual machine resource ID to assign the configuration to. Required if filter is not provided. If resourceId is provided, filter is ignored. If provided, the module scope must be set to the resource group of the virtual machine.')
+param resourceId string?
+
+@description('Conditional. Properties of the dynamic configuration assignment. Required if resourceId is not provided.')
+param filter filterType?
+
+// =============== //
+//   Deployments   //
+// =============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.maintenance-configurationassignment.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource vm 'Microsoft.Compute/virtualMachines@2024-07-01' existing = if (resourceId != null) {
```

Comment:
Will it always be in the same RG & subscription as the VM?

Reply:
If not, I'd expect a state statement

Reply:
If resourceId is provided, the module scope must be set to the resource group of the virtual machine. That's what mentioned in the resourceId description. Adding the scope to the resource vm existing does not allow to then assign vm scope to the next resource block

Reply:
![alt text](image.png)

Reply:
Strange. Will take another look tomorrow. This should work - and is something we do in quite a few places (e.g. CMK key vaults). Maybe it's a special case though. Will resolve right after 🙂